packages/atclient/src/atclient_get_public_key.c

```diff
@@ -81,16 +81,17 @@ int atclient_get_public_key(atclient *atclient, atclient_atkey *atkey, char **va
    * 5. Parse response
    */
   char *response = (char *)recv;
-  if (!atclient_string_utils_starts_with(response, "data:")) {
+  char *response_trimmed = NULL;
+  // below method points the response_trimmed variable to the position of 'data:' substring
+  if(atclient_string_utils_get_substring_position(response, DATA_TOKEN, &response_trimmed) != 0) {
     ret = 1;
     atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "recv was \"%.*s\" and did not have prefix \"data:\"\n",
                  (int)recv_len, recv);
     goto exit;
   }
+  response_trimmed = response_trimmed + 5; // +5 to skip the "data:" prefix
```

Comment:
something we could do here is

`response_trimmed = response_trimmed + strlen(DATA_TOKEN)` for future-proofing sake.

This would be optimized by the compiler to be changed to `5`, so it is not unoptimal, but it is whatever.

Reply:
Your solution looks better as it would make the code more readable, let me update that in the code

Reply:
This change has been made in [2795073](https://github.com/atsign-foundation/at_c/pull/385/commits/279507393dc0f61b49b9e177c30cef390bc3145b)